lib/pages/saved_places.dart

```diff
@@ -149,7 +154,7 @@ class SavedPlacesPageState extends State<SavedPlacesPage> {
   }
 
   Future<void> _showCurrentRoute(TrufiLocation toLocation) async {
-    final HomePageStateData dataRoute = HomePageStateData();
+    final MapRouteState dataRoute = MapRouteState();
```

Comment:
https://github.com/trufi-association/trufi-core/pull/420/files#r618910843 (here error in method _showCurrentRoute)
Before, we updated the "StateHomePage", so every time the HomePage is built "_fetch" is called (in initState method), (it was my mistake, but I needed to add code without modifying the HomePage), now we can add a (`fetch (TrufiLocation to, TrufiLocation from)`) to the HomePageCubit and add this method here.

Reply:
I am not completely sure if I understand correctly. I figured out that the MapRouteState was not immutable. Now I changed that too, and I hope it should already fix the behaviour.

Reply:
Excuse me, I did not explain it well.

Reply:
It should work like this(It is very slow in the emulator)

https://user-images.githubusercontent.com/42806689/115844334-bbe73300-a3ed-11eb-8292-fc77da9fd752.mov


but It should work like this(bug)

https://user-images.githubusercontent.com/42806689/115813154-a576b280-a3c0-11eb-80f8-3983d7b24997.mov

Reply:
This should now work again properly. I needed to take the fetch method into the HomePageCubit. I added a lot of todo´s because the fetch method is a beast. 

If we use the fetch correctly, we can also get rid of the other problem.